data-analytics/dataflow-python-examples/tests/test_data_generator.py

```diff
@@ -0,0 +1,168 @@
+        an instance of the DataGenerator class and returns fake json record that abides to the
+        rules specified in the attributes of the DataGenerator instance. Note that this is a
+        non-deterministic function so the best we can do is spot-check values obey the rules
+        for this call for thoroughness we could run the unit test many many times.
+        """
+        faker_schema = self.fakerowgen.data_gen.get_faker_schema()
+        actual_row = self.fakerowgen.generate_fake(faker_schema)
+
+        # Check returns a dict representing a single record.
+        self.assertIsInstance(actual_row, dict)
+
+        # # Check the schema is correct.
+        # self.assertEquals(actual_row.keys(), faker_schema.keys())
+
+        # Check the date in range.
+        self.assertGreaterEqual(
+            datetime.datetime.strptime(actual_row[u'lo_orderdate'], '%Y-%m-%d').date(),
+            self.data_gen.min_date)
+        self.assertLessEqual(
+            datetime.datetime.strptime(actual_row[u'lo_orderdate'], '%Y-%m-%d').date(),
+            self.data_gen.max_date)
+
+        # Check the integer is in range.
+        self.assertLessEqual(actual_row[u'lo_linenumber'], self.data_gen.max_int)
+
+        # Check the float is in range.
+        self.assertLessEqual(actual_row[u'lo_tax'], self.data_gen.max_float)
+
+        # Check int strictly positive
+        self.assertGreaterEqual(actual_row[u'lo_linenumber'], 0)
+
+        # Check float strictly positive
+        self.assertGreaterEqual(actual_row[u'lo_tax'], 0.0)
+
+        # Check key formatting.
+        r = re.compile(r'^\d{4}$')
+        self.assertTrue(r.match(actual_row[u'lo_part_key']) is not None)
+
+    def test_get_field_dict(self):
+        """
+        This tests the ability of the FakeRowGen.get_field_dict method to extract a single field
+        dictionary from a FakeRowGen.data_gen.schema
+        """
+        expected_field_dict = {u'type': u'DATE', u'name':u'lo_orderdate', u'mode': u'NULLABLE'}
+        actual_field_dict = self.fakerowgen.get_field_dict(field_name=u'lo_orderdate')
+        self.assertDictEqual(actual_field_dict, expected_field_dict)
+
+    def test_sanity_check(self):
+        fschema = self.data_gen.get_faker_schema()
+        schema_faker = FakerSchema()
+        data = schema_faker.generate_fake(fschema, 1)  # Generate one record.
+
+        # Note at this point data[u'lo_orderdate'] is a datetime.date object while Biguery expects
+        # a string
+        self.assertIsInstance(data[u'lo_orderdate'], datetime.date)
+
+        data = self.fakerowgen.sanity_check(record=data, fieldname=u'lo_orderdate')
+
+        # Check that the date was converted to a string
+        self.assertIsInstance(data[u'lo_orderdate'], unicode)
+
+        # Check that the date is in the correct format
+        _ = datetime.datetime.strptime(data[u'lo_orderdate'], '%Y-%m-%d')
```

Comment:
`_ =` not required.

This also doesn't check anything unless you're expecting an exception to be thrown and cause the test to fail, which is unclear.

Reply:
Removing unnecessary assignment.